DSharpPlus.Commands/Processors/TextCommands/ContextChecks/TextChannelTypesCheck.cs

```diff
@@ -0,0 +1,37 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+
+using DSharpPlus.Commands.ArgumentModifiers;
+using DSharpPlus.Commands.ContextChecks.ParameterChecks;
+using DSharpPlus.Entities;
+
+namespace DSharpPlus.Commands.Processors.TextCommands.ContextChecks;
+
+/// <summary>
+/// Implements a check for channel types on text commands.
+/// </summary>
+internal sealed class TextChannelTypesCheck : IParameterCheck<ChannelTypesAttribute>
+{
+    /// <inheritdoc/>
+    [SuppressMessage("Quality", "IDE0046", Justification = "Double nested ternary.")]
+    public ValueTask<string?> ExecuteCheckAsync
+    (
+        ChannelTypesAttribute attribute,
+        ParameterCheckInfo info,
+        CommandContext context
+    )
+    {
+        if (info.Value is not DiscordChannel channel)
+        {
+            return ValueTask.FromResult<string?>(null);
+        }
+
+        if (attribute.ChannelTypes.Contains(channel.Type))
```

Comment:
`SuppressMessage` won't be needed here if we use an `else if`

Reply:
same IL

Reply:
Two less lines. Feel free to merge anyways, I approved on the PR